src/App.tsx

```diff
@@ -1,54 +1,108 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+
+const debounce = (func:React.Dispatch<React.SetStateAction<string>>,
+  delay:number) => {
```

Comment:
```suggestion
const debounce = (
  func: React.Dispatch<React.SetStateAction<string>>,
  delay: number,
) => {
```
I think it's better to move this function into a separate file and import here, so you can reuse it later in other components